gems/sorbet-runtime/lib/types/compatibility_patches.rb

```diff
@@ -36,3 +36,27 @@ def observe!(method_name)
     end
   end
 end
+
+module T
```

Comment:
Would you mind adding a short comment here explaining that methods defined via sigs look like `foo(*args)` methods via reflection, and describe how to use `prepend` to enable this patch in their code, for example like this:

```ruby
require 'sorbet-runtime'
::Method.prepend(MethodExtensions)
```